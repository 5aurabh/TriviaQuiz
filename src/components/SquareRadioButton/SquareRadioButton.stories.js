import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import SquareRadioButton from './index.js';

storiesOf('SquareRadioButton', module)
  .add('default', () => <SquareRadioButton />)
  .add('with custom label', () => <SquareRadioButton label="Milk" />)
  .add('with default checked state', () => (
    <SquareRadioButton label="I am already checked" checked={true} />
  ))
  .add('with custom change handler', () => (
    <SquareRadioButton label="Click me" onChange={action('change handler')} />
  ));
