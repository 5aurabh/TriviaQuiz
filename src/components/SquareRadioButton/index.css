.SquareRadioButtonContainer {
  display: inline-block;
  margin: 0 5px;
}

.squareRadioButton {
  visibility: hidden;
  height: 0;
  width: 0;
}

.SquareRadioButtonContainer {
  display: inline-block;
}

.squareRadioButton + label {
  display: table-cell;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  background-color: #454545;
  color: white;
  padding: 15px 20px;
  border-radius: 3px;
  transition: all 0.3s ease-out;
  font-family: sans-serif;
  font-size: 0.78rem;
}

.squareRadioButton:checked + label {
  background-color: #58ba83;
}
