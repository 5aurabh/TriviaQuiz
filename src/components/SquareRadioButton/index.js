// @flow
import React from 'react';
import styles from './index.css';
type Props = {
  checked: boolean,
  name: ?string,
  id: ?string,
  label: ?string,
  value: ?string,
  onChange: () => *,
};
const SquareRadioButton = (props: Props) => {
  return (
    <div className={styles.SquareRadioButtonContainer}>
      <input
        type="radio"
        id={props.id}
        name={props.name}
        value={props.value}
        defaultChecked={props.checked}
        className={styles.squareRadioButton}
        onChange={props.onChange}
      />
      <label for={props.id}>{props.label}</label>
    </div>
  );
};

/* Add default props */
SquareRadioButton.defaultProps = {
  checked: false,
  id: 'id-0',
  name: 'name-0',
  label: 'Yes',
  value: 'some-val',
  onChange: () => {},
};

export default SquareRadioButton;
