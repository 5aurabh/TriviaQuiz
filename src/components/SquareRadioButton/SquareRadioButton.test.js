import React from 'react';
import renderer from 'react-test-renderer';
import SquareRadioButton from './index.js';

it('renders Squareradiobutton correctly', () => {
  const component = renderer.create(<SquareRadioButton />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
