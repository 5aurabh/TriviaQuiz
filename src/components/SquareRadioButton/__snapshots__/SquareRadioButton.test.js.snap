// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Squareradiobutton correctly 1`] = `
<div
  className={undefined}
>
  <input
    className={undefined}
    defaultChecked={false}
    id="id-0"
    name="name-0"
    onChange={[Function]}
    type="radio"
    value="some-val"
  />
  <label
    htmlFor="id-0"
  >
    Yes
  </label>
</div>
`;
